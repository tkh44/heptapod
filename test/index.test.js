/* eslint-disable no-sparse-arrays */
import React from 'react'
import renderer from 'react-test-renderer'
import { sheet, flush } from 'emotion'

import createResponsiveCss from '../src/index'

const css = createResponsiveCss([
  '@media(min-width: 420px)',
  '@media(min-width: 920px)',
  '@media(min-width: 1120px)',
  '@media(min-width: 11200px)'
])

describe('heptapod', () => {
  afterEach(() => flush())
  test('basic', () => {
    const cls3 = css`
      font-size: 16px;
      background: rgba(45, 213, 47, 0.11);
      color: aquamarine;
    ``
      background-color: hotpink;
    ``
      font-size: 16px;
      background: rgba(0, 0, 0, 0.11);
    `

    const tree = renderer
      .create(<div className={cls3.toString()}>Basic</div>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })
})
