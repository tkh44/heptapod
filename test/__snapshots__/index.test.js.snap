// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heptapod basic 1`] = `
.glamor-0 {
  font-size: 16px;
  background: rgba(45,213,47,0.11);
  color: aquamarine;
}

@media (min-width:420px) {
  .glamor-0 {
    background-color: hotpink;
  }
}

@media (min-width:920px) {
  .glamor-0 {
    font-size: 16px;
    background: rgba(0,0,0,0.11);
  }
}

<div
  className="glamor-0"
>
  Basic
</div>
`;

exports[`heptapod basic 2`] = `
".css-zkome7 {
  font-size: 16px;
  background: rgba(45,213,47,0.11);
  color: aquamarine;
}

.css-txi5oi {
  background-color: hotpink;
}

.css-126my4u {
  font-size: 16px;
  background: rgba(0,0,0,0.11);
}

.css-1u159q8 {
  font-size: 16px;
  background: rgba(45,213,47,0.11);
  color: aquamarine;
}

@media (min-width:420px) {
  .css-1u159q8 {
    background-color: hotpink;
  }
}

@media (min-width:920px) {
  .css-1u159q8 {
    font-size: 16px;
    background: rgba(0,0,0,0.11);
  }
}"
`;
